### ISSUE TITLE
VAE loss and classification accuracy

### ISSUE THREAD

**Issue by yuyingg0921** [hugochan/KATE#18]
hi :)   sorry to bother you again
I tried train_vae.py to train a vae model without competition layer but got high loss (60~70).
(earlystopping happened in about 20~50 epoch. I already tried a larger patience.)
The loss includes kld and reconstruction error but is this loss value in normal scale?
and then I used pred_vae.py to predict 20news document embeddings.
but after running run_classifier.py, I can only get about 22% classification accuracy.

Do you have any suggestion!?
thank you

**Comment by hugochan** [comment]
@yuyingg0921 You should be able to get reasonably good results using VAE. Can you share your hyper-parameters when training the model?

**Comment by yuyingg0921** [comment]
yes. most parameters are default value.
batch size =100
validation size = 1000
epoch = 100
dimensions = [128,128] , [512/512]
I also tried with and without a competition layer.

train_vae.py:
`python train_vae.py -i output_20news/train.corpus -nd 128 128 -sm model_vae`
![image](https://user-images.githubusercontent.com/33712951/52028895-d5b9a880-254b-11e9-8cac-a7c6c192b690.png)
with larger earlystopping patience:
![image](https://user-images.githubusercontent.com/33712951/52028928-ff72cf80-254b-11e9-9a40-003b7b45b4f6.png)


run_classifier.py:
![image](https://user-images.githubusercontent.com/33712951/52028826-7cea1000-254b-11e9-9f8f-6d5f2ef366e3.png)
![image](https://user-images.githubusercontent.com/33712951/52028836-86737800-254b-11e9-8e10-2fc2bc0d7285.png)

**Comment by Somayeh12** [comment]
Hi,
I run your code with 20newsgroup, I get the earlystopping too. it stops at epoch 33/100 and the loss is 0.48.
Any suggestion?
